src/app/workflow/version-modal/version-modal.service.ts

```diff
@@ -47,34 +47,51 @@ export class VersionModalService {
   /**
    * Saves the version.  This contains 4 parts:
    * 1. PUT workflowVersions
-   * 2. Refresh workflow
    * 3. Modify test parameter files
-   * 4. Refresh workflow again
-   * TODO: Skip 2 and 3 if there's no test parameter files to modify
+   * 4. Refresh workflow again if workflow path has changed or if a file was added/removed
    *
    * @param {WorkflowVersion} workflowVersion
    * @param {any} originalTestParameterFilePaths
    * @param {any} newTestParameterFiles
    * @memberof VersionModalService
    */
-  saveVersion(workflowVersion: WorkflowVersion, originalTestParameterFilePaths, newTestParameterFiles, workflowMode: String) {
+  saveVersion(
+    originalVersion: WorkflowVersion,
+    workflowVersion: WorkflowVersion,
+    originalTestParameterFilePaths,
+    newTestParameterFiles,
+    workflowMode: String
+  ) {
     const message1 = 'Saving workflow version';
     const message2 = 'Modifying test parameter files';
     const workflowId = this.workflowQuery.getActive().id;
+    let toRefresh = false;
+    // Checks if the workflow path was changed
+    if (originalVersion.workflow_path !== workflowVersion.workflow_path) {
+      toRefresh = true;
+    }
     this.alertService.start(message1);
     if (workflowMode !== 'HOSTED') {
       this.workflowsService.updateWorkflowVersion(workflowId, [workflowVersion]).subscribe(
         response => {
           this.alertService.start(message2);
           this.modifyTestParameterFiles(workflowVersion, originalTestParameterFilePaths, newTestParameterFiles).subscribe(
```

Comment:
Not clear to me, will this refresh if the test file path is changing but the number of files is constant?

Reply:
`4. Refresh workflow again if workflow path has changed or if a file was added/removed` and `// Checks if there was a file added/removed`

Reply:
hmm...maybe i should say specifically say test parameter file